File: CSharp/Easy/104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int MaxDepth(TreeNode no_raiz_da_arvore) {
        if(no_raiz_da_arvore == null){
            return 0;
        }

        int profundidade_do_lado_esquerdo = MaxDepth(no_raiz_da_arvore.left);

        int profundidade_do_lado_direito = MaxDepth(no_raiz_da_arvore.right);

        int maior_profundidade_entre_lados = Math.Max(profundidade_do_lado_direito,profundidade_do_lado_esquerdo);

        return 1 + maior_profundidade_entre_lados;
    }
}

File: Python/Easy/104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, no_raiz_da_arvore: Optional[TreeNode]) -> int:
        if no_raiz_da_arvore is None:
            return 0

        profundidade_do_lado_esquerdo = self.maxDepth(no_raiz_da_arvore.left)

        profundidade_do_lado_direito = self.maxDepth(no_raiz_da_arvore.right)

        maior_profundidade_entre_os_lados = (max(profundidade_do_lado_esquerdo, profundidade_do_lado_direito))

        return 1 + maior_profundidade_entre_os_lados